src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1188,18 +1188,21 @@ public void OpenFileFromPath(string xmlPath, bool forceManualExecutionMode = fal
                         OnWorkspaceOpening(xmlDoc);
 
                         // TODO: #4258
-                        // Remove this ResetEngine call when multiple home workspaces is supported.
-                        // This call formerly lived in DynamoViewModel
-                        ResetEngine();
-
-                        // TODO: #4258
                         // The following logic to start periodic evaluation will need to be moved
                         // inside of the HomeWorkspaceModel's constructor.  It cannot be there today
                         // as it causes an immediate crash due to the above ResetEngine call.
                         var hws = ws as HomeWorkspaceModel;
-                        if (hws != null && hws.RunSettings.RunType == RunType.Periodic)
+                        if (hws != null)
                         {
-                            hws.StartPeriodicEvaluation();
+                            // TODO: #4258
+                            // Remove this ResetEngine call when multiple home workspaces is supported.
+                            // This call formerly lived in DynamoViewModel
+                            ResetEngine();
```

Comment:
Opening a `CustomWorkspaceModel` prior to your fix will cause a `ResetEngine`, which calls `HomeWorkspaceModel.ResetEngine(...)`. Inside `HomeWorkspaceModel.ResetEngine` method a run is triggered for `RunType == RunType.Automatic` case. I think I agree with your assessments that this is redundant for opening custom node, just want to make sure I have not misunderstood your assessments.